finality-aleph/src/testing/data_store.rs

```diff
@@ -442,6 +448,38 @@ async fn sends_block_request_on_missing_block() {
     .await;
 }
 
+#[tokio::test]
+async fn sends_justification_request_when_not_finalized() {
+    run_test(|mut test_handler| async move {
+        let blocks = test_handler
+            .initialize_single_branch(MAX_DATA_BRANCH_LEN * 10)
+            .await;
+        test_handler.import_branch(blocks.clone()).await;
+
+        let blocks_branch = blocks[2..3].to_vec();
+        let test_data: TestData = vec![aleph_data_from_blocks(blocks_branch)];
+        test_handler.send_data(test_data.clone());
```

Comment:
you don't use `test_data` anywhere else, so there is no need to clone it, yes? also type ascription shouldn't be needed as you are passing it to `send_data` which expects `TestData`